pandas/core/indexes/multi.py

```diff
@@ -2733,7 +2733,7 @@ def intersection(self, other):
         other_tuples = other._ndarray_values
         uniq_tuples = sorted(set(self_tuples) & set(other_tuples))
         if len(uniq_tuples) == 0:
-            return MultiIndex(levels=[[]] * self.nlevels,
+            return MultiIndex(levels=self.levels,
```

Comment:
why are you changing this?

Reply:
@jreback IMHO, if the resulting intersection is empty, the levels should be preserved.  Here's a simple example:
```
mi = pd.MultiIndex.from_product([['a','b','c'],[1,2]], names=['ll','nn'])
print(mi.drop(mi))
print(mi.intersection([]))
```
This produces (with v0.22.0):
```
MultiIndex(levels=[['a', 'b', 'c'], [1, 2]],
           labels=[[], []],
           names=['ll', 'nn'])
MultiIndex(levels=[[], []],
           labels=[[], []],
           names=['ll', 'nn'])
```
Doing this change as I documented it makes the set algebra consistent.